frontend/src/components/RatingTable.tsx

```diff
@@ -0,0 +1,100 @@
+import { FC, CSSProperties, useRef, useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { VariableSizeList } from 'react-window';
+import classNames from 'classnames';
+import { Taskrating } from '../redux/roadmaps/types';
+import { FilterTypes } from '../utils/TaskUtils';
+import { titleCase } from '../utils/string';
+import { BusinessIcon, WorkRoundIcon } from './RoleIcons';
+import colors from '../colors.module.scss';
+import css from './RatingTable.module.scss';
+
+const classes = classNames.bind(css);
+
+const numFormat = new Intl.NumberFormat(undefined, {
+  minimumFractionDigits: 1,
+  maximumFractionDigits: 1,
+});
+
+export type RatingRow = FC<{
+  rating: Taskrating;
+  style?: CSSProperties;
+}>;
+
+interface RatingTableDef {
+  type: 'value' | 'work';
+  Row: RatingRow;
+}
+
+type RatingTableProps = {
+  ratings: Taskrating[];
+  avg: number;
+  searchFilter?: FilterTypes;
+  rowHeight?: number;
+  height?: number;
+};
+
+export const ratingTable: (def: RatingTableDef) => FC<RatingTableProps> = ({
+  Row,
+  type,
+}) => ({ ratings, avg, height = 500 }) => {
+  const listRef = useRef<VariableSizeList<any> | null>(null);
+  const [divRef, setDivRef] = useState<HTMLDivElement | null>(null);
+  const { t } = useTranslation();
+  const [scrollBarWidth, setScrollBarWidth] = useState(0);
+  const [rowHeights, setRowHeights] = useState<number[]>([]);
+
+  useEffect(() => {
+    if (!divRef) return;
+    setRowHeights(
+      ratings.map(({ comment }) => {
+        if (comment.length === 0) return type === 'work' ? 60 : 70;
+
+        divRef.textContent = comment;
+        const textHeight = divRef.offsetHeight;
+        divRef.textContent = '';
+
+        return textHeight + (type === 'work' ? 90 : 110);
+      }),
+    );
+    listRef.current!.resetAfterIndex(0);
+  }, [ratings, divRef]);
+
+  return (
+    <div className={classes(css.ratingContainer)}>
+      <div className={classes(css.titleContainer)}>
+        <h2>{t('Ratings title', { type: titleCase(type) })}</h2>
+      </div>
+      <div
+        style={{ marginRight: scrollBarWidth }}
+        className={classes(css.subtitleContainer)}
+      >
+        <div>{t('Average with current sorting', { type })}</div>
+        <div className={classes(css.value)}>
+          {type === 'value' ? (
+            <BusinessIcon color={colors.black100} size="small" />
+          ) : (
+            <WorkRoundIcon color={colors.black100} size="small" />
+          )}
+          {numFormat.format(avg)}
+        </div>
+      </div>
+      <hr />
+      <VariableSizeList
+        ref={listRef}
+        itemSize={(idx) => (rowHeights.length ? rowHeights[idx] : 0)}
+        itemCount={ratings.length}
+        height={height}
```

Comment:
If there's only a few ratings, it would be nice if the height could get smaller. I tried the following:
```
height={Math.min(
          height,
          rowHeights.reduce((a, b) => a + b, 0),
        )}
```
However, somehow this will make the scrollBarWidth more than zero (even with no scrollbar present) and leads to unnecessary margin. 
If anyone knows how to fix this it would be appreciated.